pkg/labels/labels.go

```diff
@@ -32,82 +32,74 @@ func HasLabelWithValue(component Labeled, labelKey string, labelValue string) bo
 	return GetLabel(component, labelKey) == labelValue
 }
 
-// SetWatchedByK8ssandraCluster sets the required labels for making a component watched by the K8ssandraCluster, i.e. a
-// modification of the component will trigger a reconciliation loop in K8ssandraClusterReconciler.
-// klusterKey specifies the namespace and name of the K8ssandraCluster.
-func SetWatchedByK8ssandraCluster(component Labeled, klusterKey client.ObjectKey) {
-	AddLabel(component, k8ssandraapi.K8ssandraClusterNameLabel, klusterKey.Name)
-	AddLabel(component, k8ssandraapi.K8ssandraClusterNamespaceLabel, klusterKey.Namespace)
+// LabelSet represents a set of labels keys and values that are applied to a K8ssandra component.
+type LabelSet map[string]string
```

Comment:
One downside of this implementation is that we allocate an extra map every time for dynamic sets. For example:
```go
labels.WatchedByK8ssandraCluster(kcKey).ApplyTo(actualPerNodeConfig)
```
Whereas a call to `SetWatchedByK8ssandraCluster` did not allocate any map previously.